exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -529,6 +549,36 @@ public long getRowCount() {
 
   }
 
+
+  // Create and return a new file selection based on reading the metadata cache file.
+  // This function also initializes a few of ParquetGroupScan's fields as appropriate.
+  private FileSelection
+  initFromMetadataCache(DrillFileSystem fs, FileSelection selection) throws IOException {
+    FileStatus metaRootDir = selection.getFirstPath(fs);
+    Path metaFilePath = new Path(metaRootDir.getPath(), Metadata.METADATA_FILENAME);
+
+    // get (and set internal field) the metadata for the directory by reading the metadata file
+    this.parquetTableMetadata = Metadata.readBlockMeta(fs, metaFilePath.toString());
+    List<String> fileNames = Lists.newArrayList();
+    for (Metadata.ParquetFileMetadata file : parquetTableMetadata.getFiles()) {
+      fileNames.add(file.getPath());
+    }
+    // when creating the file selection, set the selection root in the form /a/b instead of
+    // file:/a/b.  The reason is that the file names above have been created in the form
+    // /a/b/c.parquet and the format of the selection root must match that of the file names
+    // otherwise downstream operations such as partition pruning can break.
+    final Path metaRootPath = Path.getPathWithoutSchemeAndAuthority(metaRootDir.getPath());
+    this.selectionRoot = metaRootPath.toString();
+
+    // Use the FileSelection constructor directly here instead of the FileSelection.create() method
+    // because create() changes the root to include the scheme and authority; In future, if create()
+    // is the preferred way to instantiate a file selection, we may need to do something different...
+    FileSelection newSelection = new FileSelection(selection.getStatuses(fs), fileNames, metaRootPath.toString());
```

Comment:
Unfortunately, trying to fix this will introduce a performance regression, see [DRILL-4380](https://issues.apache.org/jira/browse/DRILL-4380) for more details.